### PR TITLE
Moved all tests to unit tests, removed Makefile, improved README for dependencies and how to test.

### PR DESCRIPTION
**Pull request by facundobatista**

### CODE REVIEW

**Comment by facundobatista** [comment]
> I'm happy to have the test suite in python test files instead of shell scripts.
> But dropping Makefile entirely doesn't seem to be a common Python pattern, and people often have an expectation that they can get it and just run 'make'. (It is also shorter than ./run_tests :)

Four keystrokes: `./r<tab>` :p.

Now seriously: I wanted to name it `test` but we have the testS directory already named like that (not in plural!), and I didn't want to push that change *now*.

I see three reasons to drop Makefile (beyond having to have `make` installed):

- it's much more natural and discoverable to run tests passing parameters: `make test ARGS="-v test.test_model.TestModelBackend"` vs `./run_tests -v test.test_model.TestModelBackend`

- it doesn't worth it for having only one target (`@python3 -m unittest`)

- even if it's a common pattern (just run `make` or `make test` or `make tests`) it opens several choices and a new developer ends up opening the Makefile to see which the options are...


> The actual time to run is about the same (make lint and python3 -m unittest test.test_infra are both around 4s for me). The autopep8 test is the bulk of that time, which is a bit unfortunate for 'unit' tests, as it is slow enough to make me not want to run the full unit test suite while iterating. But that's probably where we're at today anyway. (time ./run_tests is 12.7s)

Yes, the new way of running the tests is not slower.

And yes, autopep8 takes quite some time.

I personally don't need suggestions to fix the PEP8 issues reported by `flake8` (which is the main reason of having autopep8, right?), we probably could walk past this when the project is more stable.

File: test/test_infra.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    def test_flake8(self):
+        # verify all files are nicely styled
+        python_filepaths = get_python_filepaths()
+        style_guide = get_style_guide()
+        fake_stdout = io.StringIO()
+        with patch('sys.stdout', fake_stdout):
+            report = style_guide.check_files(python_filepaths)
+        self.assertEqual(
+            report.total_errors, 0, "There are flake8 issues!\n" + fake_stdout.getvalue())
```
Comment (jameinel) [inline test/test_infra.py]: If we feel autopep8 doesn't provide value above flake8, could we get flake8 to tell us what files to go look at and only run autopep8 on those?

Comment (facundobatista) [reply test/test_infra.py]: This is a good idea! we keep the benefits of autopep8 telling us how to fix stuff, but we speed everything up by not running it where we don't care.

**Comment by facundobatista** [comment]
> Would running `python3 setup.py test` instead of `python3 -m unittest` be better?

Don't know. Did we already have the discussion of what tests runner do we prefer? Current choice is because we selected it, or because it was the simplest thing to start? Is `pytest` out of the question?

I'll land this, all that is part of a different conversation.

**Pull request merged**